lab_3_ann_retriever/main.py

```diff
@@ -89,47 +74,62 @@ def __init__(self, stop_words: list[str]) -> None:
         Initialize an instance of the Tokenizer class.
 
         Args:
-            stop_words (list[str]): List with stop words
+            stop_words (list[str]): List with stop words.
         """
+        self._stop_words = stop_words
 
     def tokenize(self, text: str) -> list[str] | None:
         """
         Tokenize the input text into lowercase words without punctuation, digits and other symbols.
 
         Args:
-            text (str): The input text to tokenize
+            text (str): The input text to tokenize.
 
         Returns:
-            list[str] | None: A list of words from the text
+            list[str] | None: A list of words from the text.
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(text, str) or not text:
+            return None
+
+        return self._remove_stop_words(sub(r'[^a-zа-яё]+', ' ', text.lower()).split())
 
     def tokenize_documents(self, documents: list[str]) -> list[list[str]] | None:
         """
         Tokenize the input documents.
 
         Args:
-            documents (list[str]): Documents to tokenize
+            documents (list[str]): Documents to tokenize.
 
         Returns:
-            list[list[str]] | None: A list of tokenized documents
+            list[list[str]] | None: A list of tokenized documents.
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(documents, list) or not documents \
+                or not all(isinstance(doc, str) for doc in documents):
+            return None
+
+        return [tok_doc for doc in documents if (tok_doc := self.tokenize(doc)) is not None] or None
```

Comment:
do you know what ```:=``` is?

Reply:
yes, it is a walrus operator which allows me to assign a value to a variable within an expression. also it seems to be the only thing that can solve mypy error

Reply:
Nice! 🥇